app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/KeyboardWidget.java

```diff
@@ -109,6 +114,11 @@
     private boolean mInternalDeleteHint = false;
     private Session mSession;
     private boolean mInputRestarted = false;
+    public AtomicInteger mPopUpHoverDeviceId = new AtomicInteger(-1);
+    public AtomicInteger mLanguageHoverDeviceId = new AtomicInteger(-1);
+    public AtomicInteger mDomainHoverDeviceId = new AtomicInteger(-1);
+    public AtomicInteger mAutoCompleteHoverDeviceId = new AtomicInteger(-1);
+    public AtomicInteger mControlHoverDeviceId = new AtomicInteger(-1);
```

Comment:
Are these accessed from different threads? If not, they don't need to be atomic I think.

Reply:
I only did that to be able to pass the value by reference but it doesn't make much sense, I've updated it.

Reply:
I think if you do need to pass something that doesn't need to be accessed from multiple thread, the `Integer` type will work. Atomic adds some overhead. We have some code in the `onLoadRequest()` using `AtomicInteger`s that we should probably change.